piqe_ocp_lib/api/resources/ocp_nodes.py

```diff
@@ -257,3 +257,82 @@ def execute_command_on_a_node(self, node_name, command_to_execute):
             "Command - %s - execution status:\n" "RETCODE: %s\nSTDOUT: %s\nSTDERR: %s\n", command, ret, out, err
         )
         return ret, out, err
+
+    def get_master_nodes(self):
+        """
+        Method that returns a list of master node objects
+        :param label_selector: Used to return a a list of nodes based on the provided label(s)
+        :return: V1NodeList on success. None on failure.
+        """
+        master_node_object_list = None
+        try:
+            master_node_object_list = self.get_all_nodes(label_selector="node-role.kubernetes.io/master")
+        except ApiException as e:
+            logger.error("Exception when calling method list_node: %s\n", e)
+        return master_node_object_list
+
+    def get_worker_nodes(self):
+        """
+        Method that returns a list of worker node objects
+        :param label_selector: Used to return a a list of nodes based on the provided label(s)
+        :return: V1NodeList on success. None on failure.
+        """
+        worker_node_object_list = None
+        try:
+            worker_node_object_list = self.get_all_nodes(label_selector="node-role.kubernetes.io/worker")
+        except ApiException as e:
+            logger.error("Exception when calling method list_node: %s\n", e)
+        return worker_node_object_list
+
+    def is_node_schedulable(self, node_name):
+        """
+        Return the node schedulable status
+        :param node_name:
+        :return: Node schedulable status, Either True or False
+        """
+        node_object = None
+        try:
+            node_object = self.ocp_nodes.get(name=node_name)
+            if node_object.spec.unschedulable:
+                return False
+            else:
+                return True
+        except ApiException as e:
+            logger.error("Exception encountered while determining the node schedulable status: %s\n", e)
+        return node_object
+
+    def mark_node_schedulable(self, node_name):
+        """
+        Return node which was marked as schedulable
+        :param node_name:
+        :return: Name of the node which was marked as schedulable
+        """
+        api_response = None
+        body = {
+            "spec": {
+                "unschedulable": False
+            }
+        }
+        try:
+            api_response = self.ocp_nodes.patch(name=node_name, body=body)
+        except ApiException as e:
+            logger.error("Exception encountered while marking node as schedulable: %s\n", e)
+        return api_response
+
+    def mark_node_unschedulable(self, node_name):
```

Comment:
```suggestion
    def mark_node_unschedulable(self, node_name: str) -> Optional[str]:
```
please also update the docstrings of these methods to mention they could return `None`